File: 1.two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const mem = new Map();
    for (const [index, num] of nums.entries()) {
        const complement = target - num;
        if (Array.from(mem.keys()).includes(complement)) {
            return [mem.get(complement), index];
        }
        mem.set(num, index);
    }
};

File: 1.two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mem = {}
        for i, num in enumerate(nums):
            remainder = target - num;
            if remainder in mem.keys():
                return (mem[remainder], i);
            
            mem[num] = i;

        
